### PR TITLE
Add module l10n_de_holidays

### PR DESCRIPTION
**Pull request by yweng8111**
This Pull request adds module:
l10n_de_holidays
This module extends the hr_holidays_public module
and implements the functionality needed to generate and copy
the public holidays in Germany.

Dependency:
[ ] [hr_holiday_public_generator](https://github.com/OCA/hr/pull/476)

### CODE REVIEW

**Comment by ntodorova** [comment]
Hi @yweng8111,
I think everything is ok with the oca_dependencies.txt, but we have included the code in the hr_holidays_public o the generator module does not exists in the repo any more. I think if you change the dependency here, everything will be fine.

File: l10n_de_holidays/__manifest__.py
Comment (HaGuesto) [inline l10n_de_holidays/__manifest__.py]: You might want to change the website link to something more specific for your generator.

File: l10n_de_holidays/tests/__init__.py
Comment (HaGuesto) [inline l10n_de_holidays/tests/__init__.py:4]: empty line

**Review by HaGuesto — APPROVED** [review]
Thanks @yweng8111! Looks like nice work.

I tested it and also reviewed the code. It works and the code looks clean and nice!

I did not know that all the holidays can be that easily calculated. Nice!

File: l10n_de_holidays/__manifest__.py
Comment (yweng8111) [reply l10n_de_holidays/__manifest__.py]: Thanks for the Review.

File: oca_dependencies.txt
Comment (fkantelberg) [inline oca_dependencies.txt]: The dependency should point to a OCA repository. If there is another PR needed for this it should be added to the PR description.

File: l10n_de_holidays/__manifest__.py
Comment (fkantelberg) [inline l10n_de_holidays/__manifest__.py]: Avoid empty keys.

**Review by fkantelberg — CHANGES_REQUESTED** [review]
Code Review. Looks good besides the mentioned things

File: oca_dependencies.txt
Comment (yweng8111) [reply oca_dependencies.txt]: The dependant PR was added in description. Shall I change the Line 
"hr https://github.com/initOS/hr.git 11.0-hr_holidays_public_generator" into "hr" now or after the dependant PR is merged?

Comment (fkantelberg) [reply oca_dependencies.txt]: Before because this PR can only be merged when the other PR is done.

Comment (yweng8111) [reply oca_dependencies.txt]: But it looks poor if the tests are failed. No one will give a Review on it or test it with runbot. Or do you mean that I should get start with the full dependence. After successful Test change it back to the simple Repo-name.

**Comment by agent-z28** [comment]
now it looks clean to me.

**Review by agent-z28 — APPROVED** [review]

**Comment by agent-z28** [comment]
a discussed further features would be:
a) zip dependant holidays (BAVARIA) Mariä Himmelfahrt
https://www.statistik.bayern.de/statistik/bevoelkerungsstand/00141.php
b) ir.cron to autocreate the holidays few years in advance.

**Comment by pedrobaeza** [comment]
@alexey-pelykh maybe you can review this one.

**Review by fkantelberg — APPROVED** [review]
Looks good

**Comment by yweng8111** [comment]
Hi @pedrobaeza 
Could you please merge this PR?

**Comment by pedrobaeza** [comment]
We need a review from a German PSC member.

**Comment by agent-z28** [comment]
It's great work, but in my opinion, there should be an option to create all holidays automatically (optional +3 years in the future). 
Else you have to define and teach someone at the client howto handle it.
This process can easily be automated.

**Comment by yweng8111** [comment]
Actually, I have already added the guideline in Description (I mean the Usage in Readme). Maybe It should be more detailed. I will try to get the help from my friend, who can better English. I don't think it is necessary to create all holidays for 3 years. After that the user also has to learn how to generate the Holidays for the 4th year. Your last suggestion about the ir.cron was a good idea. I will try to improve the code in next version with this option. But It should be implemented in a third module and will be installed, if user activate it in the module configuration.

**Comment by agent-z28** [comment]
@pedrobaeza hello pedro who is a good German PSC member for this?

File: l10n_de_holidays/__manifest__.py
Comment (pedrobaeza) [inline l10n_de_holidays/__manifest__.py]: Please remove this key, as you already have README

Comment (yweng8111) [reply l10n_de_holidays/__manifest__.py]: @pedrobaeza Thanks for the review. Sorry about delayed correction. I was in holidays last Month.

**Comment by pedrobaeza** [comment]
@yweng8111 there is this problem in current repo:
```
./l10n_de_tax_statement/i18n/de.po:108: keyword "Aktualisieren" unknown
./l10n_de_tax_statement/i18n/de.po:108:195: syntax error
./l10n_de_tax_statement/i18n/de.po:1035: keyword "gesendet" unknown
./l10n_de_tax_statement/i18n/de.po:1035:200: syntax error
```

**Comment by astirpe** [comment]
Hi @yweng8111, can you fix the errors directly in this PR? Just deleting the error lines would be fine.

**Comment by yweng8111** [comment]
@astirpe Done

**Comment by yweng8111** [comment]
@pedrobaeza Could you please try to merge it again?

**Comment by pedrobaeza** [comment]
It still seems not enough:
```
./l10n_de_tax_statement_zm/i18n/de.po:169: end-of-line within string
msgmerge: found 1 fatal error
```

**Pull request merged**
